music_assistant/server/providers/deezer/helpers.py

```diff
@@ -0,0 +1,506 @@
+                )
+            ],
+        )
+    except AttributeError:
+        return MediaItemMetadata(
+            preview=track.preview,
+        )
+
+
+async def parse_metadata_album(album: deezer.Album) -> MediaItemMetadata:
+    """Parse the album metadata."""
+    return MediaItemMetadata(
+        images=[MediaItemImage(type=ImageType.THUMB, path=album.cover_big)],
+    )
+
+
+async def parse_metadata_artist(artist: deezer.Artist) -> MediaItemMetadata:
+    """Parse the artist metadata."""
+    return MediaItemMetadata(
+        images=[MediaItemImage(type=ImageType.THUMB, path=artist.picture_big)],
+    )
+
+
+async def _get_album(mass, track: deezer.Track) -> Album | None:
+    try:
+        return await parse_album(mass=mass, album=await get_album_from_track(track=track))
+    except AttributeError:
+        return None
+
+
+async def get_album_from_track(track: deezer.Track) -> deezer.Album:
+    """Get track's artist."""
+
+    def _get_album_from_track():
+        try:
+            return track.get_album()
+        except deezer.exceptions.DeezerErrorResponse:
+            return None
+
+    return await asyncio.to_thread(_get_album_from_track)
+
+
+async def get_artist_from_track(track: deezer.Track) -> deezer.Artist:
+    """Get track's artist."""
+
+    def _get_artist_from_track():
+        return track.get_artist()
+
+    return await asyncio.to_thread(_get_artist_from_track)
+
+
+async def get_artist_from_album(album: deezer.Album) -> deezer.Artist:
+    """Get track's artist."""
+
+    def _get_artist_from_album():
+        return album.get_artist()
+
+    return await asyncio.to_thread(_get_artist_from_album)
+
+
+async def get_albums_by_artist(artist: deezer.Artist) -> deezer.PaginatedList:
+    """Get albums by an artist."""
+
+    def _get_albums_by_artist():
+        return artist.get_albums()
+
+    return await asyncio.to_thread(_get_albums_by_artist)
+
+
+async def get_artist_top(artist: deezer.Artist) -> deezer.PaginatedList:
+    """Get top tracks by an artist."""
+
+    def _get_artist_top():
+        return artist.get_top()
+
+    return await asyncio.to_thread(_get_artist_top)
+
+
+async def parse_track(mass, track: deezer.Track) -> Track:
+    """Parse the deezer-python track to a MASS track."""
+    artist = await get_artist_from_track(track=track)
+    return Track(
+        item_id=str(track.id),
+        provider=mass.domain,
+        name=track.title,
+        media_type=MediaType.TRACK,
+        sort_name=track.title_short,
+        position=track.track_position,
+        duration=track.duration,
+        artists=[await parse_artist(mass=mass, artist=artist)],
+        album=(await _get_album(mass=mass, track=track)),
+        provider_mappings={
+            ProviderMapping(
+                item_id=str(track.id),
+                provider_domain=mass.domain,
+                provider_instance=mass.instance_id,
+                content_type=ContentType.MP3,
```

Comment:
The quality is now hardcoded to Mp3 which is probably not right for Deezer where you can even have lossless flac tracks. So the contenttype should be set to the correct codec here, retrieved from the full track details.

Reply:
Yeah it was just temporary. Because dzr only outputted mp3

Reply:
Fixed in newest commit